cdap-data-fabric/src/test/java/co/cask/cdap/data/format/CSVStringsRecordFormatTest.java

```diff
@@ -0,0 +1,208 @@
+    Schema mapSchema = Schema.mapOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.STRING));
+    Schema schema = Schema.recordOf("event", Schema.Field.of("f1", Schema.arrayOf(mapSchema)));
+    FormatSpecification formatSpec =
+      new FormatSpecification(DelimitedStringsRecordFormat.class.getCanonicalName(),
+                              schema, Collections.<String, String>emptyMap());
+
+    DelimitedStringsRecordFormat format = new CSVRecordFormat();
+    format.initialize(formatSpec);
+  }
+
+
+  @Test(expected = UnsupportedTypeException.class)
+  public void testMapFieldInvalid() throws UnsupportedTypeException {
+    Schema mapSchema = Schema.mapOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.STRING));
+    Schema schema = Schema.recordOf("event", Schema.Field.of("f1", mapSchema));
+    FormatSpecification formatSpec =
+      new FormatSpecification(DelimitedStringsRecordFormat.class.getCanonicalName(),
+                              schema, Collections.<String, String>emptyMap());
+
+    DelimitedStringsRecordFormat format = new CSVRecordFormat();
+    format.initialize(formatSpec);
+  }
+
+  @Test(expected = UnsupportedTypeException.class)
+  public void testRecordFieldInvalid() throws UnsupportedTypeException {
+    Schema recordSchema = Schema.recordOf("record", Schema.Field.of("recordField", Schema.of(Schema.Type.STRING)));
+    Schema schema = Schema.recordOf("event", Schema.Field.of("f1", recordSchema));
+    FormatSpecification formatSpec =
+      new FormatSpecification(DelimitedStringsRecordFormat.class.getCanonicalName(),
+                              schema, Collections.<String, String>emptyMap());
+
+    DelimitedStringsRecordFormat format = new CSVRecordFormat();
+    format.initialize(formatSpec);
+  }
+
+  @Test
+  public void testStringArrayFormat() throws UnsupportedTypeException, UnexpectedFormatException {
+    DelimitedStringsRecordFormat format = new CSVRecordFormat();
+    format.initialize(null);
+    String body = "userX,actionY,itemZ";
+    StructuredRecord output = format.read(ByteBuffer.wrap(Bytes.toBytes(body)));
+    String[] actual = output.get("body");
+    String[] expected = body.split(",");
+    Assert.assertTrue(Arrays.equals(expected, actual));
+  }
+
+  @Test
+  public void testDelimiter() throws UnsupportedTypeException, UnexpectedFormatException {
+    DelimitedStringsRecordFormat format = new CSVRecordFormat();
+    FormatSpecification spec = new FormatSpecification(DelimitedStringsRecordFormat.class.getCanonicalName(),
+                                                       null,
+                                                       ImmutableMap.of(DelimitedStringsRecordFormat.DELIMITER, " "));
+    format.initialize(spec);
+    String body = "userX actionY itemZ";
+    StructuredRecord output = format.read(ByteBuffer.wrap(Bytes.toBytes(body)));
+    String[] actual = output.get("body");
+    String[] expected = body.split(" ");
```

Comment:
hm I agree it would be better to have csv and tsv just map to different settings.  Will change RecordFormats